neuralmonkey/runners/label_runner.py

```diff
@@ -96,3 +60,50 @@ def collect_results(self, results: List[Dict]) -> None:
             scalar_summaries=None,
             histogram_summaries=None,
             image_summaries=None)
+
+class LabelRunner(BaseRunner):
+
+    def __init__(self,
+                 output_series: str,
+                 decoder: Any,
+                 postprocess: Callable[[List[str]], List[str]] = None
+                ) -> None:
+        super(LabelRunner, self).__init__(output_series, decoder)
+        self._postprocess = postprocess
+
+        # make sure the lazy decoder creates its output tensor
+        log("Decoder output tensor: {}".format(decoder.decoded))
+
+    def get_executable(self,
+                       compute_losses: bool = False,
+                       summaries: bool = True,
+                       num_sessions: int = 1) -> LabelRunExecutable:
+        if compute_losses:
+            if not hasattr(self._decoder, "cost"):
+                raise TypeError("Decoder should have the 'cost' attribute")
+            fetches = {"loss": getattr(self._decoder, "cost")}
+        else:
+            fetches = {}
+
+        if not hasattr(self._decoder, "logprobs"):
+            raise TypeError("Decoder should have the 'logprobs' attribute")
+
+        if not hasattr(self._decoder, "encoder"):
+            raise TypeError("Decoder should have the 'encoder' attribute")
+
+        if not hasattr(self._decoder, "vocabulary"):
+            raise TypeError("Decoder should have the 'vocabulary' attribute")
+
+        fetches["label_logprobs"] = getattr(self._decoder, "logprobs")
+        fetches["input_mask"] = getattr(self._decoder,
+                                        "encoder").input_sequence.mask
```

Comment:
když bude union tak k tomu můžeš přistoupit normálně přes tečku. I teď bys to tak asi moh dělat a jen psát něco jako `# type: ignore` na konec řádku.